dot/network/block_announce.go

```diff
@@ -214,31 +214,36 @@ func (s *Service) validateBlockAnnounceHandshake(peer peer.ID, hs Handshake) err
 
 	bestBlockNum := big.NewInt(int64(bhs.BestBlockNumber))
 
+	np, ok := s.notificationsProtocols[BlockAnnounceMsgType]
```

Comment:
Wherever we are accessing `notificationsProtocols` we should lock it.
```
	s.notificationsMu.Lock()
	defer s.notificationsMu.Unlock()
```

Take a look at `dot/network/service.go:394`

Same for `handshakeData` map.